File: T-Bank/Algorithms-and-Data-Structures-2025/13sprint-Mathematics/contest/A-Шестерёнки/solution.cpp
#include <iostream>


long long gcd(long long x, long long y) {
    if (x == 0) return y;
    return gcd(y % x, x);
}



long long lcm(long long x, long long y) {
    return x * y / gcd(x, y);
}


void solution() {
    /*
    Time Complexity O(log(N))
    Memory Complexity O(log(N))
    */
    long long n, k;
    std::scanf("%lld %lld", &n, &k);
    std::printf("%lld\n", lcm(n, k));
}


int main() {
    solution();
}

File: T-Bank/Algorithms-and-Data-Structures-2025/13sprint-Mathematics/contest/A-Шестерёнки/solution.py
import sys


def gcd(x: int, y: int) -> int:
    if x == 0: return y
    return gcd(y % x, x)


def lcm(x: int, y: int) -> int:
    return x * y // gcd(x, y)


def solution() -> None:
    """
    Time Complexity O(log(N))
    Memory Complexity O(log(N))
    """
    n, k = map(int, sys.stdin.readline().rstrip().split())
    sys.stdout.write('{}\n'.format(lcm(n, k)))


if __name__ == '__main__':
    solution()